src-ghc/Pact/ApiReq.hs

```diff
@@ -41,22 +43,21 @@ import Data.Text.Encoding
 import Data.Thyme.Clock
 import Data.Default (def)
 
-import Crypto.Ed25519.Pure
-
 import Pact.Types.Crypto
 import Pact.Types.Util
 import Pact.Types.Command
 import Pact.Types.RPC
 import Pact.Types.Runtime hiding (PublicKey)
 import Pact.Types.API
 
-
-data KeyPair = KeyPair {
-  _kpSecret :: PrivateKey,
-  _kpPublic :: PublicKey
-  } deriving (Eq,Show,Generic)
-instance ToJSON KeyPair where toJSON = lensyToJSON 3
-instance FromJSON KeyPair where parseJSON = lensyParseJSON 3
+data ApiKeyPair = ApiKeyPair {
+  _akpSecret  :: PrivateKeyBS,
+  _akpPublic  :: Maybe PublicKeyBS,
+  _akpAddress :: Maybe Text,
+  _akpScheme  :: Maybe PPKScheme
+  } deriving (Eq, Show, Generic)
+instance ToJSON ApiKeyPair where toJSON = lensyToJSON 4
```

Comment:
What context will these JSON serializations be used?  Will they be persisted in a way that requires backwards compatibility?  If so, then we might consider writing the JSON instances by hand to avoid upstream changing it on us without warning.

Reply:
They are used in `pact -a` and in the PactContinuationSpec. It essentially makes it easier to format API request JSON using a YAML file describing the request.

See: https://pact-language.readthedocs.io/en/latest/pact-reference.html#api-request-formatter

Reply:
Ok, this is probably not a big deal.